### ISSUE TITLE
CORS Error trying to update Whitelist

### ISSUE THREAD

**Issue by guice** [hassio-addons/addon-pi-hole#126]
# Problem/Motivation

I'm getting a CORS error trying to update the Whitelist, even though the Addon Admin screen still works.

## Expected behavior

The hostname get added

## Actual behavior

CORS error:
<img width="1324" alt="Screen Shot 2020-03-02 at 8 21 05 AM" src="https://user-images.githubusercontent.com/205378/75690220-b534b400-5c5f-11ea-9206-6ed853b85c6a.png">

My configuration:

```yaml
update_lists_on_start: false
ssl: true
certfile: ven[redacted]
keyfile: ven[redacted]
interface: ''
ipv6: true
ipv4_address: ''
ipv6_address: ''
hosts:
  - name: [redacted].local
    ip: 192.168.0.88
  - name: ven[redacted]
    ip: 192.168.0.88
  - name: [redacted].local
    ip: 192.168.0.69
virtual_host: ven[redacted]
```

Might be related to: https://github.com/hassio-addons/addon-pi-hole/pull/90

**Comment by debsahu** [comment]
I am also experiencing the same CORS error. The workaround is I connect https://192.1268.0.XXX:8123 and add things to whitelist. Hope that the solution is simple and something that we missed.

**Comment by guice** [comment]
The workaround does work - although it's fails certificate validation (understandably). It's not a proper solution, though.

**Comment by sinclairpaul** [comment]
I can't duplicate this with the default config: 
`update_lists_on_start: false
ssl: true
certfile: fullchain.pem
keyfile: privkey.pem
interface: ''
ipv6: true
ipv4_address: ''
ipv6_address: ''
hosts: []`

and the latest version.

As per the issue template it would be great if you could include the required logs in any issue logged.

**Comment by guice** [comment]
Logs for what? It's a CORS error. Are you saying there are JS capture client-side errors? CORS technically isn't an error, though. It's like a 4xx request for browsers.

**Comment by sinclairpaul** [comment]
Logs so we can determine the platform you run and the add-on version..

**Comment by guice** [comment]
> the platform you run and the add-on version..

Oh. It wasn't in the New Issue template. That's easily:

* Home Assistant on NUC PC, Docker environment. Version 1.106.2
* Pihole Addon, 4.1.0

The issue is I'm loading my site via a custom hostname. That hostname is not listed in Pihole's list of value CORS domains:

```
Failed CORS: ven[redacted] vs 192.168.0.88, , a0d7b954-pi-hole, pi.hole, localhost, [servername], [server dn]
```

Says "ven[redacted]" is not in list: ip address, pihold addon hostname, "pi.hole", localhost, server hostname, and server dn

How do I add "ven[redacted]" into the list of valid hosts? It was taken out in #90.

**Comment by MakkaKaplar** [comment]
Just installed this addon and I'm having the same problem when saving settings.
`https://xxxxx.duckdns.org/hassio/ingress/a0d7b954_pi-hole`

`Failed CORS: xxxxx.duckdns.org vs 192.168.1.4, , a0d7b954-pi-hole, pi.hole, localhost, hassio, hassio.local`

**Comment by frenck** [comment]
Fixed in v4.1.1
